src/App.tsx

```diff
@@ -14,11 +15,108 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  isGoodsVisible: boolean,
+  goods: string[],
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    isGoodsVisible: false,
+    goods: [...goodsFromServer],
+  };
+
+  showGoodsList = () => {
+    this.setState({ isGoodsVisible: true });
+  };
+
+  reverseGoods = () => {
+    this.setState((state) => {
+      return {
+        goods: [...state.goods].reverse(),
```

Comment:
reverseGoods = () => {
    this.setState((state) => ({
     goods: [...state.goods].reverse(),
    }));
  };

Reply:
I agree, but in my opinion it is so obvious that we are returning the object

Reply:
Agree with @OleksandrNi . Will be less code